pkg/reconcilers/deploymentconfig.go

```diff
@@ -54,9 +55,23 @@ func GenericBackendMutators() []DCMutateFn {
 		DeploymentConfigPriorityClassMutator,
 		DeploymentConfigTopologySpreadConstraintsMutator,
 		DeploymentConfigPodTemplateAnnotationsMutator,
+		DeploymentConfigEnvMutator,
```

Comment:
Since we are not doing this for listener, personally, instead of adding it to both, listener, cron and worker and then checking if dealing with worker proceed otherwise dont, I would add the mutator here instead:
https://github.com/3scale/3scale-operator/blob/master/pkg/3scale/amp/operator/backend_reconciler.go#L66

```
workerConfigMutator = append(workerConfigMutator, reconcilers.DeploymentConfigReplicasMutator, reconcilers.DeploymentConfigEnvMutator)
```
and remove the IF statement.

Reply:
Think I'll change it up, I added the if as I didn't want manually applied ENV getting reverted on other backend components. But this is not needed using the individual mutator.

Reply:
Hmm doesn't reconcile back the ENV when removed so I will have to stick to the original method

Reply:
Thanks, we figured out why ( related to https://github.com/3scale/3scale-operator/pull/784 )

Reply:
Ok so that logic only gets hit if you have replicas set for backendworker in the apimanager CR. Will leave as is